arch/arm/core/swap_helper.S

```diff
@@ -342,65 +370,8 @@ _thread_irq_disabled:
      */
     bx lr
 
-#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
-
-/**
- *
- * @brief Service call handler
- *
- * The service call (svc) is used in the following occasions:
- * - IRQ offloading
- * - Kernel run-time exceptions
- *
- * @return N/A
- */
-SECTION_FUNC(TEXT, z_arm_svc)
-    /* Use EXC_RETURN state to find out if stack frame is on the
-     * MSP or PSP
-     */
-    ldr r0, =0x4
-    mov r1, lr
-    tst r1, r0
-    beq _stack_frame_msp
-    mrs r0, PSP
-    bne _stack_frame_endif
-_stack_frame_msp:
-    mrs r0, MSP
-_stack_frame_endif:
-
-    /* Figure out what SVC call number was invoked */
-    ldr r1, [r0, #24]   /* grab address of PC from stack frame */
-    /* SVC is a two-byte instruction, point to it and read  encoding */
-    subs r1, r1, #2
-    ldrb r1, [r1, #0]
-
-   /*
-    * grab service call number:
-    * 1: irq_offload (if configured)
-    * 2: kernel panic or oops (software generated fatal exception)
-    * Planned implementation of system calls for memory protection will
-    * expand this case.
-    */
-
-    cmp r1, #2
-    beq _oops
-
-#if defined(CONFIG_IRQ_OFFLOAD)
-    push {r0, lr}
-    bl z_irq_do_offload  /* call C routine which executes the offload */
-    pop {r0, r1}
-    mov lr, r1
-#endif /* CONFIG_IRQ_OFFLOAD */
-
-    /* exception return is done in z_arm_int_exit() */
-    b z_arm_int_exit
-
-_oops:
-    push {r0, lr}
-    bl z_do_kernel_oops
-    pop {r0, pc}
-
-#elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
+#if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE) || \
```

Comment:
@andrewboie check here: now there's a common SVC implementation for all Cortex-M variants.